js/bundle.go

```diff
@@ -346,6 +347,14 @@ func (b *Bundle) instantiate(vuImpl *moduleVUImpl, vuID uint64) (*BundleInstance
 
 	<-initDone
 
+	if err == nil {
+		var finished bool
+		bi.mainModule, finished, err = result.Result()
+		if !finished {
+			return nil, errors.New("initializing the main module hasn't finished, this is a bug in k6 please report it")
```

Comment:
Are there any other essential details we could include in the message to help us identify the case when needed? Like error, something else, maybe even panic to have a stack trace? What do you think?

Reply:
I would expect in all cases we will need the test to actually figure this out. It will likely be some very strange case as it will need to have the runnign the things on thevent loop to:
1. not have errored out
2. not have finished everything

The stacktrace here is ... very likely pointless as it wil just show us which executors is used, as the thing that hasn't finished is alredy past.

We can panic :shrug: I am just not into panicking all that much and again - I couldn't figure out a case where this can happen. This is here mostly just in case the unthinkable happens.

But I am okay with panicking if that seems better.

Reply:
If panic doesn't help, we shouldn't. 

My point is that, if possible, we can print out some details about the state that could be included in future bug reports, which could be handy for our future investigations.

Reply:
There isn't really any info at this point - we started to load modules, after the event loop had nothing else to run, the modules were still not loaded.

This will require us to get the test and see how it runs to figure it out.